src/dev/build.ts

```diff
@@ -0,0 +1,42 @@
+import { ServerContext } from "../server/context.ts";
+import { FreshOptions, Manifest } from "../server/mod.ts";
+import { dirname, fromFileUrl, join } from "../server/deps.ts";
+import { fs } from "./deps.ts";
+
+export async function build(
+  manifestPath: string,
+  opts: FreshOptions,
+) {
+  const manifest = (await import("file://" + manifestPath)).default as Manifest;
+
+  const outDir = join(dirname(fromFileUrl(manifest.baseUrl)), "_fresh");
+
+  // Ensure that build dir is empty
+  await fs.emptyDir(outDir);
+
+  const ctx = await ServerContext.fromManifest(manifest, {
+    ...opts,
+    loadSnapshot: false,
+  });
+
+  // Bundle assets
+  const snapshot = await ctx.buildSnapshot();
+
+  // Write output files to disk
+  await Promise.all(snapshot.paths.map((fileName) => {
+    const data = snapshot.read(fileName);
+    if (data === null) return;
+
+    return Deno.writeFile(join(outDir, fileName), data);
+  }));
+
+  // Write dependency snapshot file to disk
+  const deps: Record<string, string[]> = {};
+  for (const filePath of snapshot.paths) {
+    const dependencies = snapshot.dependencies(filePath);
+    deps[filePath] = dependencies;
+  }
```

Comment:
Is the ordering here deterministic? Ie. will no change cause a potential snapshot file change?

Reply:
We could always sort the paths to ensure determinism.